Add new error code NU5037 to the docs (#1619)
NuGet/docs.microsoft.com-nuget: docs/reference/errors-and-warnings/NU5037.md

---
title: NuGet Error NU5037
description: NU5037 Error code
author: kartheekp-ms
ms.author: karann
ms.date: 8/9/2019
ms.topic: reference
ms.reviewer: anangaur
f1_keywords: 
  - NU5037
---

# NuGet Error NU5037
<pre>The package is missing the required nuspec file.</pre>

### Issue

Restore fails when the [nuspec](../nuspec.md) file is missing from the source feed or global cache folder of the package.

## Scenario 1

> nuspec file is missing from the source feed of the package.

Reviewer: ```suggestion
nuspec file is missing from the package (nupkg) file.
```

### Solution

Contact package author.

## Scenario 2

Packages are managed using Package Reference format.

### Solution

Delete the package folder specified in the error message and run the restore again.

## Scenario 3

> Packages are managed using Packages.config format.

Reviewer 2: Consider 
```suggestion
Restore fails for a project managed with the packages.config format.
```

### Solution

> Package path is not appended to the error message. Identifying the package with missing nuspec file requires manual probing of dependencies. Delete corrupted package folder or entire packages folder if the package with missing nuspec file is unknown and run the restore again.

Reviewer: ```suggestion
Package path is not appended to the error message. Identifying the package with missing nuspec file requires manual probing of dependencies. Delete corrupted package folder or entire solution packages folder if the package with missing nuspec file is unknown and run the restore again.
```

> Package path is not appended to the error message. Identifying the package with missing nuspec file requires manual probing of dependencies. Delete corrupted package folder or entire solution packages folder if the package with missing nuspec file is unknown and run the restore again.

Reviewer 2: I would remove 
> Package path is not appended to the error message

The rest is fine.
